tests/multiprocess_gpu_test.py

```diff
@@ -209,5 +281,251 @@ def test_gpu_multi_node_initialize_and_psum(self):
     self.assertEqual(y[0], jax.device_count())
     print(y)
 
+  # TODO: sudhakarsingh27 to change/omit test in favor of using `Array`
+  # since `GlobalDeviceArray` is going to be deprecated in the future
+  def test_pjit_gda_multi_input_multi_output(self):
+    jax_multiprocess_init()
+    global_mesh = jtu.create_global_mesh((8, 2), ("x", "y"))
+    global_input_shape = (16, 2)
+    global_input_data = np.arange(
+        util.prod(global_input_shape)).reshape(global_input_shape)
+
+    def cb(index):
+      return global_input_data[index]
+
+    mesh_axes1 = experimental.PartitionSpec("x", "y")
+    gda1 = global_device_array.GlobalDeviceArray.from_callback(
+        global_input_shape, global_mesh, mesh_axes1, cb)
+    mesh_axes2 = experimental.PartitionSpec("x")
+    gda2 = global_device_array.GlobalDeviceArray.from_callback(
+        global_input_shape, global_mesh, mesh_axes2, cb)
+    mesh_axes3 = experimental.PartitionSpec(("x", "y"))
+    gda3 = global_device_array.GlobalDeviceArray.from_callback(
+        global_input_shape, global_mesh, mesh_axes3, cb)
+
+    with parallel_functions_output_gda(True):
```

Comment:
Please set this up in setUp once!

There's no need to use a context manager since all the tests are running with GDA

Reply:
Since all the tests run in their own subprocess, there can't be a `setUp` for all the tests. (Do you mean doing this at the start/end of each test though?)

Reply:
What do you mean by there can't be a setUp for all the tests?

The setUp method exists on all test classes. Can you try prototyping it please?

```
class Slurm...():
  
  def setUp(self):
    super().setUp()
    self.gda_enabled = jax.config.parallel_functions_output_gda
    jax.config.update('parallel_functions_output_gda', True)

  def tearDown(self):
    jax.config.update('parallel_functions_output_gda', self.gda_enabled)
    super().tearDown()
```

Reply:
Ah that's clearer (doing setUp/tearDown for every test and not once for all the tests). I can try prototyping this